src/MinterGateway.sol

```diff
@@ -768,17 +774,13 @@ contract MinterGateway is IMinterGateway, ContinuousIndexing, ERC712Extended {
 
         if (missedIntervals_ == 0) return;
 
-        // Save until when the minter has been penalized for missed intervals to prevent double penalizing them.
-        minterState_.penalizedUntilTimestamp = missedUntil_;
-
-        uint112 principalOfActiveOwedM_ = principalOfActiveOwedMOf(minter_);
-
-        if (principalOfActiveOwedM_ == 0) return;
-
         uint112 penaltyPrincipal_ = _imposePenalty(minter_, uint152(principalOfActiveOwedM_) * missedIntervals_);
 
         if (penaltyPrincipal_ == 0) return;
 
+        // Save until when the minter has been penalized for missed intervals to prevent double penalizing them.
+        minterState_.penalizedUntilTimestamp = missedUntil_;
```

Comment:
Setting `minterState_.penalizedUntilTimestamp` before `_imposePenalty` is critical because the penalized until timestamp needs to be captured even if the penalty rate is 0.

Reply:
I would not call it critical but moved back above.